src/trajectory_data_handling/trajectory_sql.cpp

```diff
@@ -96,16 +100,43 @@ namespace trajectory_data_handling {
                 std::cout << "Error in switch statement in spatial_query_on_rtree_table." << '\n';
         }
 
-        query << "SELECT id FROM " << table_name << " WHERE minLongitude<=" << get<1>(longitudeRange) << " AND maxLongitude>=" << get<0>(longitudeRange) << " AND minLatitude<=" << get<1>(latitudeRange) << " AND maxLatitude>=" << get<0>(latitudeRange) << " AND minTimestamp<=" << get<1>(timestampRange) << " AND maxTimestamp>=" << get<0>(timestampRange) << ";";
+        query << "SELECT id FROM " << table_name << " WHERE minLongitude<=" << window.x_high << " AND maxLongitude>=" << window.x_low << " AND minLatitude<=" << window.y_high << " AND maxLatitude>=" << window.y_low << " AND minTimestamp<=" << window.t_high << " AND maxTimestamp>=" << window.t_low << ";";
         trajectory_data_handling::query_handler::run_sql(query.str().c_str(), purpose);
 
+
         if (!trajectory_data_handling::query_handler::trajectory_ids_in_range.empty()){
             switch(purpose) {
                 case query_purpose::load_original_rtree_into_datastructure:
                     load_database_into_datastructure(query_purpose::load_original_trajectory_information_into_datastructure,trajectory_data_handling::query_handler::trajectory_ids_in_range);
+                    for(int i = 0; i < trajectory_data_handling::query_handler::original_trajectories->size(); ) {
+                        auto &trajectory = (*trajectory_data_handling::query_handler::original_trajectories)[i];
+                        if (trajectory.id != 0) {
+                            auto keep_trajectory = spatial_queries::Range_Query::in_range(trajectory, window);
+                            if (!keep_trajectory) {
+                                trajectory_data_handling::query_handler::original_trajectories->erase(trajectory_data_handling::query_handler::original_trajectories->begin() + i);
+                            } else {
+                                ++i;
+                            }
+                        } else {
+                            ++i;
+                        }
+                    }
```

Comment:
Kan dette for-loop ikke laves om til en funktion for at undgå code-duplication? Det samme loop bruges både til simplified og original trajectories.